code/modules/modular_computers/os/_os.dm

```diff
@@ -309,4 +309,20 @@
 /datum/extension/interactive/os/proc/mail_received(datum/computer_file/data/email_message/received)
 	var/datum/computer_file/program/email_client/e_client = locate() in running_programs
 	if(e_client)
-		e_client.mail_received(received)
\ No newline at end of file
+		e_client.mail_received(received)
+
+/datum/extension/interactive/os/proc/run_script(mob/user, var/datum/computer_file/data/script)
+	open_terminal(user)
+	var/datum/terminal/T = has_terminal(user)
+	if(!istype(T))
+		return  TOPIC_HANDLED
+
+	T.show_terminal(user)
+	T.append_to_history(">Running '[script.filename].[script.filetype]'")
+	var/list/lines = splittext(script.stored_data, "\[br\]")
```

Comment:
Are newlines in the script automatically converted to pencode, or do you have to explicitly write `[br]` to separate lines in scripts?

Reply:
Looks like it may be? When you input it, it replaces newlines with `[br]` tags, and then when you edit it, it replaces the [br] tags with newlines. Good enough for me.